src/util/mapbox.js

```diff
@@ -113,5 +129,13 @@ function parseUrl(url: string): UrlObject {
 
 function formatUrl(obj: UrlObject): string {
     const params = obj.params.length ? `?${obj.params.join('&')}` : '';
-    return `${obj.protocol}://${obj.authority}${obj.path}${params}`;
+    const url = `${obj.protocol}://${obj.authority}${obj.path}${params}`;
+    return transformURL(url);
 }
+
+exports.resolveURL = function(url: string): string {
+    url = transformURL(url);
+    const a = window.document.createElement('a');
+    a.href = url;
+    return transformURL(a.href);
```

Comment:
Do we need to go through `a.href` again with the transformed URL? Or only with the transformed URL?